engine/core/src/main/java/org/datacleaner/util/LabelUtils.java

```diff
@@ -96,13 +97,26 @@ public static String getLabel(ComponentJob job) {
      * 
      * @return
      */
-    public static String getLabel(ComponentJob job, boolean includeDescriptorName, boolean includeInputColumnNames,
+    public static String getLabel(ComponentJob job, boolean includeDescriptorName, boolean includeInputColumnNames, boolean includeRequirements) {
+        return getLabel(job, null, includeDescriptorName, includeInputColumnNames, includeRequirements);
+    }
+
+    public static String getLabel(ComponentJob job, AnalyzerResult result, boolean includeDescriptorName, boolean includeInputColumnNames,
             boolean includeRequirements) {
         final String jobName = job.getName();
         final StringBuilder label = new StringBuilder();
         if (Strings.isNullOrEmpty(jobName)) {
             ComponentDescriptor<?> descriptor = job.getDescriptor();
-            label.append(descriptor.getDisplayName());
+            String displayName = descriptor.getDisplayName();
+            if (result != null) {
+                if (result instanceof HasLabelAdvice) {
```

Comment:
The solution with getName sounds nice to me if there are not too many implementing classes. Will look into that. It would also avoid the sudden updating of the tab name that currently happens. Also, for the record, I blame all misdirection on the hints provided by someone in #468 .